src/rpc/rawtransaction.cpp

```diff
@@ -336,6 +339,120 @@ static UniValue gettxchainlocks(const JSONRPCRequest& request)
     return result_arr;
 }
 
+static UniValue getassetunlockchainlocks(const JSONRPCRequest& request)
+{
+    RPCHelpMan{
+            "getassetunlockchainlocks",
+            "\nReturns the status of given Asset Unlock indexes.\n",
+            {
+                    {"indexes", RPCArg::Type::ARR, RPCArg::Optional::NO, "The Asset Unlock indexes (no more than 100)",
+                     {
+                             {"index", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "An Asset Unlock index"},
+                     },
+                    },
+            },
+            RPCResult{
+                    RPCResult::Type::ARR, "", "Response is an array with the same size as the input txids",
+                    {
+                            {RPCResult::Type::OBJ, "xxxx", "Asset Unlock index.",
+                             {
+                                     {RPCResult::Type::STR, "", "Status of the Asset Unlock index: {chainlocked|mempooled|null}"},
+                             }},
+                    }
+            },
+            RPCExamples{
+                    HelpExampleCli("getassetunlockchainlocks", "'[\"myindex\",...]'")
+                    + HelpExampleRpc("getassetunlockchainlocks", "[\"myindex\",...]")
+            },
+    }.Check(request);
```

Comment:
Have this in a separate method

Reply:
OK (Just no other RPC has a separate method on this file fyi)